test/test_gps.py

```diff
@@ -0,0 +1,46 @@
+import subprocess
+from unittest import mock
+
+import os
+import urllib.error
+import wget
+
+from gnssrefl.gps import *
+
+
+def test_get_sopac_navfile_working(mocker):
+    mocker.patch("wget.download")
+    mocker.patch("os.path.exists")
+    mocker.patch("subprocess.call")
+    os.path.exists.return_value = True
+    assert (
+        get_sopac_navfile("p1031050.20.snr66", "2020", "20", "105")
+        == "p1031050.20.snr66"
+    )
+    wget.download.assert_called_once_with(
+        "ftp://garner.ucsd.edu/pub/rinex/2020/105/p1031050.20.snr66.Z",
+        "p1031050.20.snr66.Z",
+    )
+    os.path.exists.assert_called_once_with("p1031050.20.snr66")
+    subprocess.call.assert_called_once_with(["uncompress", "p1031050.20.snr66.Z"])
```

Comment:
Unit test for `get_sopac_navfile` that verifies its functionality without requiring outbound network access - thus fast and guaranteed not to break in future. Does not verify that external archive is still operational; that should be verified in an integration test or smoke test, not a unit test.

We patch the functions called by `get_sopac_navfile` that would have impacts on the external world, and instruct the mock framework what the return value or other behavior should be when these functions are called. Then we invoke `get_sopac_navfile` and assert that the desired parameters were passed to the patched functions.